### ISSUE TITLE
Fetch and parse message as an `email.message.Message`

### ISSUE THREAD

**Issue by NicolasLM** [mjs/imapclient#340]
Python has a convenient type to represent email messages. We could add an alternate `fetch` method that returns it.

Actually there are two types, the current [EmailMessage](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage) and the legacy [Message](https://docs.python.org/3/library/email.compat32-message.html#email.message.Message).

**Comment by freddrake** [comment]
Is there a reason not to leave it to the application to deal with the message itself?  It's easy enough to use the convenience functions from the ``email`` package for this, and not all applications need the ``email`` package's message objects.

I'm thinking about separation of concerns here.  Yes, it could be slightly more convenient for some applications.

A counter would be that letting ``imapclient`` build the message object would be that message bodies, especially large attachments, could be delayed until the application requested them; the email structure could be built based on the BODYSTRUCTURE / HEADERS values.  But that'll be a chunk of work.

**Comment by NicolasLM** [comment]
Do not underestimate the power of sane defaults. Currently the result of a `fetch` call is slightly complicated to understand. Having a blessed Python object to represent it would in my opinion help developers which are not well versed into IMAP and email in general.

But we both agree, this is only about convenience.

**Comment by Bastien-Brd** [comment]
Hi guys, 

I agree with the above, but as a new user of IMAPClient and generally new to IMAP in general, it took me a while to understand where the scope of IMAPClient stops and where I should use another package for easily parsing the email message itself. 

All I wanted was to quickly check emails from a Gmail account and download attachment from some of them. A high level API for doing so is handy as IMAP is not a straight forward protocol.
I understand now I should use something like the standard `email` module in addition, in order to parse the messages themselves.

IMAPClient is really helpful, but maybe the README.md could be more explicit as per the scope of this tool, for the non IMAP-savvy reader.

**Comment by freddrake** [comment]
Makes sense.  I think this would be best handled as a documentation enhancement.
